contracts/automation/andromeda-evaluation/src/contract.rs

```diff
@@ -0,0 +1,1045 @@
+    use super::*;
+    use crate::mock_querier::{mock_dependencies_custom, MOCK_QUERY_CONTRACT};
+    use andromeda_automation::evaluation::Operators;
+    use common::app::AndrAddress;
+    use cosmwasm_std::testing::{mock_env, mock_info};
+
+    #[test]
+    fn test_initialization() {
+        let mut deps = mock_dependencies_custom(&[]);
+
+        let condition_address = AndrAddress {
+            identifier: "condition_address".to_string(),
+        };
+        let query_address = AndrAddress {
+            identifier: MOCK_QUERY_CONTRACT.to_string(),
+        };
+        let task_balancer = AndrAddress {
+            identifier: "task_balancer_address".to_string(),
+        };
+        let msg = InstantiateMsg {
+            condition_address,
+            query_address,
+            task_balancer,
+        };
+        let info = mock_info("creator", &[]);
+
+        // we can just call .unwrap() to assert this was a success
+        let res = instantiate(deps.as_mut(), mock_env(), info, msg).unwrap();
+        assert_eq!(0, res.messages.len());
+
+        // make sure address was saved correctly
+        let addr = CONDITION_ADO_ADDRESS.load(&deps.storage).unwrap();
+        assert_eq!(
+            addr,
+            AndrAddress {
+                identifier: "condition_address".to_string(),
+            }
+        );
+
+        let addr = QUERY_ADO_ADDRESS.load(&deps.storage).unwrap();
+        assert_eq!(
+            addr,
+            AndrAddress {
+                identifier: MOCK_QUERY_CONTRACT.to_string(),
+            }
+        )
+    }
+
+    #[test]
+    fn test_evaluate_greater_is_greater() {
+        let mut deps = mock_dependencies_custom(&[]);
+
+        let condition_address = AndrAddress {
+            identifier: "condition_address".to_string(),
+        };
+        let query_address = AndrAddress {
+            identifier: MOCK_QUERY_CONTRACT.to_string(),
+        };
+        let task_balancer = AndrAddress {
+            identifier: "task_balancer_address".to_string(),
+        };
+        let operation = Operators::Greater;
+        let msg = InstantiateMsg {
+            condition_address,
+            query_address,
+            task_balancer,
+        };
+        let info = mock_info("creator", &[]);
+
+        let _res = instantiate(deps.as_mut(), mock_env(), info.clone(), msg).unwrap();
+        // Set query result to 40
+        let user_value = Uint128::new(30);
+        let msg = ExecuteMsg::Evaluate {
+            user_value,
+            operation,
+        };
+
+        let res = execute(deps.as_mut(), mock_env(), info, msg).unwrap();
+        let expected = Response::new()
+            .add_attribute("result", "true".to_string())
+            .add_submessage(SubMsg::reply_on_error(
+                CosmosMsg::Wasm(WasmMsg::Execute {
+                    contract_addr: "condition_address".to_string(),
+                    msg: to_binary(&andromeda_automation::condition::ExecuteMsg::StoreResult {
```

Comment:
There's a few references to message types imported like this, for readability purposes we should import these and rename: `andromeda_autiomation::condition::{ExecuteMsg as ConditionExecuteMsg}`